src/client/service-components/EditableText.js

```diff
@@ -0,0 +1,97 @@
+import React, {useState} from 'react';
+import {Keys} from '../constants/constants.js';
+import EditIcon from '../images/edit-icon-18.svg';
+
+import './EditableText.css';
+
+
+export const EditableText = ({value, saveChanges, editButton = false, className, multipleRows = false}) => {
+    const [isEdit, setIsEdit] = useState(false);
+    const [isInputEmpty, setIsInputEmpty] = useState(false);
+    const [text, setText] = useState(value);
+    const [previousText, setPreviousText] = useState(value);
+
+    const handleEditClick = () => {
+        setIsEdit(true);
+    };
+
+    const updateText = ({target}) => {
+        setText(target.value);
+        setIsInputEmpty(target.value === '');
+    };
+
+    const handleKeyDown = (e) => {
+        if (e.key === Keys.Enter) {
+            setIsEdit(false);
```

Comment:
Why don't you create a separate function for this part in `if` that is absolutely the same in `handleBlur`?